agent/internal/container/container_api_test.go

```diff
@@ -215,3 +218,134 @@ func TestContainer(t *testing.T) {
 		})
 	}
 }
+
+func TestContainerStatus(t *testing.T) {
+	log.SetLevel(log.TraceLevel)
+
+	t.Log("building client")
+	rawCl, err := dclient.NewClientWithOpts(dclient.WithAPIVersionNegotiation(), dclient.FromEnv)
+	require.NoError(t, err)
+	defer func() {
+		if cErr := rawCl.Close(); cErr != nil {
+			t.Logf("closing docker client: %s", cErr)
+		}
+	}()
+	cl := docker.NewClient(rawCl)
+
+	tests := []struct {
+		name                string
+		image               string
+		entrypoint          []string
+		signalAtState       cproto.State
+		signalAtEventAction string
+		signal              syscall.Signal
+		failure             *aproto.ContainerFailure
+	}{
+		{
+			name:                "killed after start and before run",
+			image:               "ubuntu",
+			entrypoint:          []string{"echo", "hello"},
+			signalAtState:       cproto.Starting,
+			signalAtEventAction: "create",
+			signal:              syscall.SIGKILL,
+			failure: &aproto.ContainerFailure{
+				FailureType: aproto.ContainerAborted,
+				ErrMsg:      "killed before run",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			id := cproto.NewID()
+
+			// setup event listener
+			eventChannel := make(chan struct{})
+			errChannel := make(chan error)
+			go func() {
+				t.Logf("creating %s event listener for %s container", tt.signalAtEventAction, id.String())
+				filterArgs := filters.NewArgs()
+				filterArgs.Add("type", "container")
+				filterArgs.Add("event", tt.signalAtEventAction)
+				filterArgs.Add("event.Actor.ID", id.String())
+
+				eventListener, errListener := rawCl.Events(context.Background(),
+					types.EventsOptions{Filters: filterArgs})
+				timeout := time.After(10 * time.Second)
+				for {
+					select {
+					case <-eventListener:
+						t.Logf("found %s event for %s container",
+							tt.signalAtEventAction,
+							id.String(),
+						)
+						close(eventChannel)
+						return
+					case err := <-errListener:
+						errChannel <- fmt.Errorf("failed to listen for %s events for %s container: %s",
+							tt.signalAtEventAction,
+							id.String(),
+							err,
+						)
+					case <-timeout:
+						errChannel <- fmt.Errorf("timed out while listening for %s events for %s container",
+							tt.signalAtEventAction,
```

Comment:
Here and other usages of ```errChannel``` it might be simpler to just fail the test in place rather then sending it to a channel to be failed in another place

Reply:
Actually, my point doesn't make sense because if I get rid of the eventChannel & errChannel, there is not point to using that goroutine at all for the eventListener. So this gets handled with that work 🥳